### PR TITLE
#13170: Fixed Incorrect Output Shape Calculation for Dilation > 1

### PR DESCRIPTION
**Pull request by sankarmanoj-tt**
### Ticket
#13170 

### Problem description
Incorrect Output shape. 

### What's changed
Used `sliding_window.get_output_shape()` instead. 

### Checklist
- [ ] Post commit CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/11211036432)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by pavlejosipovic — COMMENTED** [review]
Seems like we should add a regression UT for this.

**Comment by sankarmanoj-tt** [comment]
> Seems like we should add a regression UT for this.

Done.

File: ttnn/cpp/ttnn/operations/conv/conv2d/device/multi_core_optimized_conv_sharded/optimized_conv_op_width_sharded_v2.cpp [hunk 1]
```diff
@@ -398,16 +398,10 @@ operation::ProgramWithCallbacks multi_core_optimized_conv_width_sharded_v2_impl(
...
-        conv_act_size_h,
-        conv_act_size_w,
-        filter_h,
-        filter_w,
-        stride_h,
-        stride_w,
-        pad_h,
-        pad_w,
-        extra_padding_for_32B_alignment);
+    auto output_shape = sliding_window_config.get_output_shape();
```
[the remark this answers is not in the document]
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/device/multi_core_optimized_conv_sharded/optimized_conv_op_width_sharded_v2.cpp:401]: AFAIK, `extra_padding_for_32B_alignment` is [hard-coded as 0.](https://github.com/tenstorrent/tt-metal/blob/7f2a2f0747933dbd6bbc2c6db5a5b35099f06e80/ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp#L797)

I think it should be removed. @mywoodstock Please confirm.

[the remark this answers is not in the document]
Comment (mywoodstock) [reply ttnn/cpp/ttnn/operations/conv/conv2d/device/multi_core_optimized_conv_sharded/optimized_conv_op_width_sharded_v2.cpp:401]: If its not used, yes, please remove.

**Review by bbradelTT — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**